apitools/org.eclipse.pde.api.tools/src/org/eclipse/pde/api/tools/internal/ApiAnalysisApplication.java

```diff
@@ -189,6 +190,31 @@ public Object start(IApplicationContext context) throws Exception {
 		}
 	}
 
+	private static void printMarker(IMarker marker, String type) {
+		String path = getFullPath(marker);
+		String file = marker.getResource().getName();
+		int lineNumber = marker.getAttribute(IMarker.LINE_NUMBER, -1);
+		String message = marker.getAttribute(IMarker.MESSAGE, "").trim(); //$NON-NLS-1$
+		String description = marker.getAttribute("description", "").trim(); //$NON-NLS-1$ //$NON-NLS-2$
+		if (!description.isEmpty()) {
+			message = String.format("%s %s", message, description); //$NON-NLS-1$
```

Comment:
Why not use String concatenation directly here and below?
IMHO this is more readable and performance whise this should probably better since Java 9/10/11.

Reply:
I don't think it matters much here, but I found it often easier to read using String.format, that's not really a performance concern here and you only ever need one NON-NLS per line ;-)